plugins/woocommerce/src/Blocks/Utils/ProductCollectionUtils.php

```diff
@@ -92,4 +92,117 @@ private static function remove_empty_array_recursive( $array ) {
 		}
 		return $array;
 	}
+
+	/**
+	 * Translate WP Post instance to context.
+	 * 
+	 * Possible contexts:
+	 * - page
+	 * - post
+	 * - product
+	 * - product-archive
+	 * - checkout
+	 * - catalog
+	 * - cart
+	 * - checkout
+	 * - order
+	 *
+	 * @param WP_Post $post The Post instance.
+	 *
+	 * @return string|false Returns the context. False for unknown or invalid context.
+	 */
+	public static function parse_editor_location_context( $post ) {
+
+		if ( ! $post instanceof \WP_Post ) {
+			return false;
+		}
+
+		$post_type = $post->post_type;
+		if ( ! in_array( $post_type, array( 'post', 'page', 'wp_template', 'wp_template_part' ), true ) ) {
+			return false;
+		}
+
+		$context = 'post';
+
+		if ( in_array( $post_type, array( 'wp_template', 'wp_template_part' ), true ) ) {
+
+			$name = $post->post_name;
+			if ( false !== strpos( $name, 'single-product' ) ) {
+				$context = 'product';			
+			} elseif ( false !== strpos( $name, 'taxonomy-' ) ) { // Including the '-' in the check to avoid false positives.
+				$taxonomy           = str_replace( 'taxonomy-', '', $name );
+				$product_taxonomies = get_object_taxonomies( 'product', 'names' );
+				if ( in_array( $taxonomy, $product_taxonomies, true ) ) {
+					$context = 'product-archive';
+				}
+			} elseif(  'page-cart' === $name ) {
+				$context = 'cart';
+			} elseif(  'page-checkout' === $name ) {
+				$context = 'checkout';
```

Comment:
Are Cart and Checkout pages guaranteed to have these slugs?

Reply:
This question applies to every slug here.

Reply:
Thanks @nerrad! 

I double-checked if there was any intention to modify them. I think it's safe to use them. However, I made a few changes to align them better with the templates registry:
https://github.com/woocommerce/woocommerce/blob/9888ee7f4bb8e3170cb7345c1a5c8603f316de35/plugins/woocommerce/src/Blocks/BlockTemplatesRegistry.php#L41-L52